common/txmgr/confirmer.go

```diff
@@ -435,6 +448,54 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Che
 	return nil
 }
 
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) ProcessStuckTransactions(ctx context.Context, blockNum int64) error {
+	// Use the detector to find a stuck tx for each enabled address
+	stuckTxs, err := ec.stuckTxDetector.DetectStuckTransactions(ctx, ec.enabledAddresses, blockNum)
+	if err != nil {
+		return fmt.Errorf("failed to detect stuck transactions: %w", err)
+	}
+	if len(stuckTxs) == 0 {
+		return nil
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(stuckTxs))
+	errorList := []error{}
+	var errMu sync.Mutex
+	for _, tx := range stuckTxs {
+		// All stuck transactions will have unique from addresses. It is safe to process separate keys concurrently
+		// NOTE: This design will block one key if another takes a really long time to execute
+		go func(tx txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) {
+			defer wg.Done()
+			lggr := tx.GetLogger(ec.lggr)
+			// Create an purge attempt for tx
+			purgeAttempt, err := ec.TxAttemptBuilder.NewPurgeTxAttempt(ctx, tx, lggr)
```

Comment:
Let us say, we were successful in broadcasting a purge attempt here.
But for gas spike reasons, this purged attempt is still not confirmed. The next iteration of processHead() will again call this method.
Is this code able to handle cases when an already existing purge attempt is there?

Or should this code just bail quickly if it sees pre-existing purge attempt?
I think it can easily assume that the pre-existing purge attempt will go through natural gas bump cycles?

Reply:
The `DetectStuckTransactions` method actually handles filtering out txs already marked for purge. So this will only run once for a tx considered to be stuck. After that the normal bumping logic would take care of getting the purge attempt confirmed. The `AttemptBuilder` also has logic to ensure that bumps for purge attempts are also empty attempts and marked for purge.